### PR TITLE
TiKV : Stale reads.: add histogram of min safe ts. 

### LINKED ISSUE #17307: add histogram of safe ts in follower
### Steps to reproduce

current max_gap_safe_ts gives only upper bound of a node. However, it doesn't provide how many regions are behind. 
### What did you expect?
added a histogram of min safe ts in follower. It is a valuable metrics to see how many regions are actually behind.

### PR DESCRIPTION
**Pull request by mittalrishabh**
### What is changed and how it works?
added a histogram of min safe ts in follower. It is a valuable metrics to see how many regions are actually behind.  current max_gap_safe_ts gives only upper bound of a node. 

Issue Number: Close #17307

What's Changed:

```commit-message

```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by mittalrishabh** [comment]
verified a change

**Review by zhangjinpeng87 — APPROVED** [review]
LGTM

File: components/resolved_ts/src/endpoint.rs [hunk 1]
```diff
@@ -419,10 +420,18 @@ where
                     }
                 } else {
                     // follower safe-ts
+                    RTS_MIN_FOLLOWER_SAFE_TS_GAP_HISTOGRAM.observe(now.saturating_sub(
```
Comment (cfzjywxk) [inline components/resolved_ts/src/endpoint.rs:423]: I don't quite understand the meaning of using `now()` to subtract the temporary variable used in the loop to calculate the minumum resolved-ts or safe-ts.

If the target is to observe the resolved-ts lag for each region, should the
```
let lag_for_region = now - read_progress_of_current_region.current_resolved_ts
histogram.observe(lag_for_region)
```
be used instead?

Comment (mittalrishabh) [reply components/resolved_ts/src/endpoint.rs:423]: yeh, this is wrong. let me fix it.

**Comment by cfzjywxk** [comment]
@mittalrishabh 
The change LGTM, should we merge this with the grafana dashboard changes, or we merge this PR first?

**Comment by mittalrishabh** [comment]
Lets merge this PR. I will add it in a separate PR.

**Comment by mittalrishabh** [comment]
> @mittalrishabh The change LGTM, should we merge this with the grafana dashboard changes, or we merge this PR first?

Lets merge this PR. I will create a new PR for grafana dashboard.

**Review by cfzjywxk — APPROVED** [review]
LGTM

**Pull request merged**